dotnet/src/Functions/Functions.OpenApi/OpenApiKernelPluginFactory.cs

```diff
@@ -291,6 +292,37 @@ async Task<RestApiOperationResponse> ExecuteAsync(Kernel kernel, KernelFunction
     /// <summary>The metadata property bag key to use for the list of extension values provided in the swagger file at the operation level.</summary>
     private const string OperationExtensionsMetadataKey = "operation-extensions";
 
+    /// <summary>
+    /// Converts operation id to valid <see cref="KernelFunction"/> name.
+    /// A function name can contain only ASCII letters, digits, and underscores.
+    /// </summary>
+    /// <param name="operation">The REST API operation.</param>
+    /// <param name="logger">The logger.</param>
+    /// <returns>Valid KernelFunction name.</returns>
+    private static string ConvertOperationToValidFunctionName(RestApiOperation operation, ILogger logger)
+    {
+        if (!string.IsNullOrWhiteSpace(operation.Id))
+        {
+            return ConvertOperationIdToValidFunctionName(operationId: operation.Id, logger: logger);
+        }
+
+        // Tokenize operation path on forward and back slashes
+        string[] tokens = operation.Path.Split('/', '\\');
+        StringBuilder result = new();
+        result.Append(CultureInfo.CurrentCulture.TextInfo.ToTitleCase(operation.Method.ToString()));
+
+        foreach (string token in tokens)
+        {
+            // Removes all characters that are not ASCII letters, digits, and underscores.
+            string formattedToken = RemoveInvalidCharsRegex().Replace(token, "");
+            result.Append(CultureInfo.CurrentCulture.TextInfo.ToTitleCase(formattedToken.ToLower(CultureInfo.CurrentCulture)));
+        }
+
+        logger.LogInformation("""Operation method "{0}" with path "{1}" converted to "{2}" to comply with SK Function name requirements. Use "{2}" when invoking function.""", operation.Method, operation.Path, result, result);
```

Comment:
nit: It's better to use names in log message parameters, since these parameters can be stored in separate columns in telemetry tool, which allows to query and filter log messages:
```suggestion
logger.LogInformation(
    """Operation method "{Method}" with path "{Path}" converted to "{Result}" to comply with SK Function name requirements. Use "{Result}" when invoking function.""",
    operation.Method,
    operation.Path,
    result,
    result);
```

Reply:
Fixed here and also elsewhere in the file